File: backend/schemas/response_envelope.py
from typing import Any, Dict, Optional

from pydantic import BaseModel


class ResponseEnvelope(BaseModel):
    status: str
    data: Optional[Dict[str, Any]] = None
    meta: Optional[Dict[str, Any]] = None
    error: Optional[Dict[str, Any]] = None

    class Config:
        extra = "forbid"

File: frontend/src/types/responseEnvelope.ts
export interface ErrorDetail {
  code?: string;
  message?: string;
  details?: any;
}

export interface ResponseEnvelope<T = any> {
  status: string;
  data: T | null;
  meta?: Record<string, any>;
  error?: ErrorDetail | null;
}

export default ResponseEnvelope;
